cosima_cookbook/database.py

```diff
@@ -380,16 +380,29 @@ def build_index(directories, session, client=None, update=False):
     session.commit()
     return indexed
 
-def delete_missing(session, ncfiles):
-    """Given a database session and a list of NCFile objects,
-    ensure the file backing each NCFile exists, else delete
-    it from the database.
+def prune_experiment(experiment, session, delete=True):
```

Comment:
Is this intended to be called manually by a user when an experiment has been deleted and the database needs updating?

Is there any point when querying a DB that it might be appropriate to call this if files are missing? Or issue a warning and the python code necessary to do this if the user does not have write permissions on the DB.

Reply:
> Is this intended to be called manually by a user when an experiment has been deleted and the database needs updating?

I didn't actually think of removing an entire experiment as an intended use case.

> Is there any point when querying a DB that it might be appropriate to call this if files are missing? Or issue a warning and the python code necessary to do this if the user does not have write permissions on the DB.

I think the appropriate time to suggest using this would be after an index which failed on some files. We wouldn't know at the moment whether a query failed due to a missing file (but xarray would complain, at least).